### PR TITLE
Primer::Alpha::ActionBar component

### PR DESCRIPTION
**Pull request by jonrohan**
### Description

https://github.com/github/primer/issues/1148

This PR upstreams our experimental ActionBar component. Design guidelines detailed here https://primer.style/design/components/action-bar

Closes # (type the issue number after # if applicable; otherwise remove this line)

### Integration

> Does this change require any updates to code in production?

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews

### CODE REVIEW

File: docs/src/@primer/gatsby-theme-doctocat/nav.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@
     url: "/status"
 - title: Components
   children:
+  - title: ActionBar - Fix my order in docs/src/@primer/gatsby-theme-doctocat/nav.yml
```
Comment (camertron) [inline docs/src/@primer/gatsby-theme-doctocat/nav.yml:18]: ?

**Review by camertron — CHANGES_REQUESTED** [review]
Looking good! Just needs some tests 👍

File: app/components/primer/alpha/action_bar.pcss [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/* CSS for ActionBar  */
+.ActionBar {
+  position: relative;
+  display: flex !important;
+  min-width: calc(var(--primer-control-medium-size, 32px) * 3);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:5]:
```suggestion
  min-width: calc(var(--control-medium-size) * 3); 
```

This took me awhile to understand. Maybe add a comment like "show minimum of 3 buttons"

File: app/components/primer/alpha/action_bar.pcss [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  position: relative;
+  flex-shrink: 0;
+}
+
+.ActionBar-more-menu {
+  flex-shrink: 0;
+}
+
+.ActionBar--small {
+  min-width: calc(var(--primer-control-small-size, 28px) * 3);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:32]:
```suggestion
  min-width: calc(var(--control-small-size) * 3);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.ActionBar--large {
+  min-width: calc(var(--primer-control-large-size, 40px) * 3);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:36]:
```suggestion
  min-width: calc(var(--control-large-size) * 3);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/* Divider */
+
+.ActionBar-divider {
+  height: calc(var(--primer-control-medium-size, 32px) / 2);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:42]:
```suggestion
  height: calc(var(--control-medium-size) / 2);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  margin: 0 var(--primer-controlStack-medium-gap-condensed, 8px);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:43]:
```suggestion
  margin: 0 var(--controlStack-medium-gap-condensed);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  border-left: var(--primer-borderWidth-thin, 1px) solid var(--color-border-subtle);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:44]:
```suggestion
  border-left: var(--borderWidth-thin) solid var(--color-border-subtle);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.ActionBar--small .ActionBar-divider {
+  margin: 0 var(--primer-controlStack-small-gap-condensed, 8px);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:48]:
```suggestion
  margin: 0 var(--controlStack-small-gap-condensed);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+.ActionBar--large .ActionBar-divider {
+  margin: 0 var(--primer-controlStack-large-gap-condensed, 8px);
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:52]:
```suggestion
  margin: 0 var(--controlStack-large-gap-condensed);
```

File: app/components/primer/alpha/action_bar.pcss [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/* Increase spacing so touch targets don't overlap */
+
+@media (pointer: coarse) {
+  .ActionBar .ActionBar-item-container {
+    gap: calc(var(--primer-control-minTarget-coarse, 44px) - var(--primer-control-medium-size, 32px)); /* 12px */
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:59]:
```suggestion
    gap: calc(var(--control-minTarget-coarse) - var(--control-medium-size)); /* 12px */
```

File: app/components/primer/alpha/action_bar.pcss [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  .ActionBar--small .ActionBar-item-container {
+    gap: calc(var(--primer-control-minTarget-coarse, 44px) - var(--primer-control-small-size, 28px));  /* 16px */
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:63]:
```suggestion
    gap: calc(var(--control-minTarget-coarse) - var(--control-small-size));  /* 16px */
```

File: app/components/primer/alpha/action_bar.pcss [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  .ActionBar--large .ActionBar-item-container {
+    gap: calc(var(--primer-control-minTarget-coarse, 44px) - var(--primer-control-large-size, 40px));  /* 4px */
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.pcss:67]:
```suggestion
    gap: calc(var(--control-minTarget-coarse) - var(--control-large-size));  /* 4px */
```

**Review by keithamus — APPROVED** [review]

File: app/components/primer/alpha/action_bar.html.erb [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<%= render Primer::BaseComponent.new(**@system_arguments) do %>
+  <%= render Primer::BaseComponent.new(tag: :div, classes: "ActionBar-item-container", "data-target": "action-bar.itemContainer") do %>
+    <% items.each do |item| %>
+      <%= item %>
+    <% end %>
+  <% end %>
+  <% if render_overflow_menu? %>
+    <%= render(@action_menu) do |menu| %>
+      <% menu.with_show_button(icon: :"kebab-horizontal", "aria-label": "Menu", size: @size, scheme: :invisible) %>
```
Comment (langermank) [inline app/components/primer/alpha/action_bar.html.erb:9]: Just wondering if the aria-label should be a param that could be passed in by the consumer? Maybe with a default of "Menu"? I'm just not sure if that's something we may want to customize for accessibility.

**Review by langermank — APPROVED** [review]

**Pull request merged**